src/hotspot/share/opto/mulnode.cpp

```diff
@@ -369,17 +369,24 @@ Node *MulLNode::Ideal(PhaseGVN *phase, bool can_reshape) {
   if (bit1 == abs_con) {           // Found a power of 2?
```

Comment:
May I suggest, since you are working on this code, maybe factoring out the idiom `x & -x`?  (Sadly, I think there may be multiple opinions about exactly where to name this idiom, which might prevent us from starting the fix.  If that's the case I suggest a follow-up bug.)

```
diff --git a/src/hotspot/share/utilities/powerOfTwo.hpp b/src/hotspot/share/utilities/powerOfTwo.hpp
index a98b81e8037..83713d373ee 100644
--- a/src/hotspot/share/utilities/powerOfTwo.hpp
+++ b/src/hotspot/share/utilities/powerOfTwo.hpp
@@ -119,4 +119,14 @@ inline T next_power_of_2(T value)  {
   return round_up_power_of_2(value + 1);
 }
 
+// Return the largest power of two that is a submultiple of the given value.
+// This is the same as the numeric value of the least-significant set bit.
+// For unsigned values, it replaces the old trick of (value & -value).
+// precondition: value > 0.
+template<typename T, ENABLE_IF(std::is_integral<T>::value)>
+inline T submultiple_power_of_2(T value) {
+  assert(value > 0, "Invalid value");
+  return value & -value;
+}
+
 #endif // SHARE_UTILITIES_POWEROFTWO_HPP
```